resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/MultiValuedArrayParamConverter.java

```diff
@@ -0,0 +1,214 @@
+      Method saveAsString = null;
+      if (converter == null && delegate == null)
+      {
+         try
+         {
+            saveAsString = values[0].getClass().getMethod("saveAsString");
+         }
+         catch (Exception e)
+         {
+            // ok
+         }  
+      }
+      char separatorChar = getSeparatorChar();
+      StringBuilder sb = new StringBuilder();
+      boolean first = true;
+      for (Object v : values)
+      {
+         if (first)
+         {
+            first = false;
+         }
+         else
+         {
+            sb.append(separatorChar);
+         }
+         if (v instanceof String)
+         {
+            sb.append(v);
+         }
+         else if (converter != null)
+         {
+            sb.append(converter.toString(v)); 
+         }
+         else if (delegate != null)
+         {
+            sb.append(delegate.toString(v));
+         }
+         else if (saveAsString != null)
+         {
+            try
+            {
+               sb.append(saveAsString.invoke(v));
+            }
+            catch (Exception e)
+            {
+               throw new RuntimeException(Messages.MESSAGES.errorCallingSaveAsString(v));
+            }  
+         }
+         else
+         {
+            sb.append(v.toString());
+         }
+      }
+      return sb.toString();
+   }
+
+   private char getSeparatorChar()
+   {
+      if (separator.charAt(0) == '[')
+      {
+         return separator.charAt(1);
+      }
+      else
+      {
+         return separator.charAt(0);
+      }
+   }
+
+   private void copyPrimitiveArray(Object[] objectArray, Object primitiveArray)
+   {
+      Class<?> primitiveClass = primitiveArray.getClass().getComponentType();
+      for (int i = 0; i < objectArray.length; i++)
+      {
+         if (boolean.class.equals(primitiveClass))
+         {
+            objectArray[i] = Array.getBoolean(primitiveArray, i);
+         }
+         else if (byte.class.equals(primitiveClass))
+         {
+            objectArray[i] = Array.getByte(primitiveArray, i);
+         }
+         else if (short.class.equals(primitiveClass))
+         {
+            objectArray[i] = Array.getShort(primitiveArray, i);
+         }
+         else if (int.class.equals(primitiveClass))
+         {
+            objectArray[i] = Array.getInt(primitiveArray, i);
+         }
+         else if (long.class.equals(primitiveClass))
+         {
+            objectArray[i] = Array.getLong(primitiveArray, i);
+         }
+         else if (char.class.equals(primitiveClass))
+         {
+            objectArray[i] = Array.getChar(primitiveArray, i);
+         }
+         else if (float.class.equals(primitiveClass))
+         {
+            objectArray[i] = Array.getFloat(primitiveArray, i);
+         }
+         else if (double.class.equals(primitiveClass))
+         {
+            objectArray[i] = Array.getDouble(primitiveArray, i);
+         }
+      }
+   }
+   
+   private boolean isHeader()
```

Comment:
Instead of iterating again on the annotations array can we better add a getter in `StringParameterInjector` to use the `paramType` already stored here ?

Reply:
@NicoNes, yes, now I'm just pulling HeaderDelegate out of the StringParameterInjector. isHeader() is gone.

Reply:
:+1: